muikku-core-plugins/src/main/resources/META-INF/resources/scripts/gui/communicator.js

```diff
@@ -82,37 +81,43 @@ $(document).ready(function(){
       var mCont = $(CommunicatorImpl.msgContainer);
       this._addLoading(CommunicatorImpl.msgContainer);
       
-      mApi().communicator.messages.read(communicatorMessageId).callback($.proxy(function (err, result) {
-        for (var i = 0; i < result.length; i++) {
-          result[i].caption = $('<div>').html(result[i].caption).text();
-          result[i].content = $('<div>').html(result[i].content).text();
+      mApi().communicator.messages
+        .read(communicatorMessageId)
+        .on("$", function (message, messageCallback) {
           
-          var sId = result[i].id;
-          mApi().communicator.communicatormessages.sender.read(sId)
-            .callback(function (err, user) {  
-
-            if (MUIKKU_LOGGED_USER_ID == user.id){
-              result[i].isOwner = true;         
-            } else {
-              result[i].isOwner = false;    
-            }               
-
-            result[i].senderFullName = user.firstName + ' ' + user.lastName;
-            result[i].senderHasPicture = user.hasImage;
+          mApi().communicator.communicatormessages.sender.read(message.id).callback(function (err, user) {  
+            if(err){
+              $('.notification-queue').notificationQueue('notification', 'error', getLocaleText('plugin.communicator.showmessage.thread.error'));
+            }else{            
+              message.isOwner = MUIKKU_LOGGED_USER_ID === user.id;
+              message.senderFullName = user.firstName + ' ' + user.lastName;
+              message.senderHasPicture = user.hasImage;
+              message.caption = $('<div>').html(message.caption).text();
```

Comment:
$('<div>').html(message.caption).text(); parses the html content and translates it into text, so i think the div should be right here instead of dust file.